### ISSUE TITLE
Add support for RealmQuery.asObservable()

### ISSUE THREAD

**Issue by cmelchior** [realm/realm-java#1955]
Subtask of #1710 
Depends on https://github.com/realm/realm-core/issues/1410 being implemented, or making sure that the Core Query class can expose all needed data.

**Comment by beeender** [comment]
@cmelchior I don't think this is something we should do. The listeners currently are based on OS collection notifier which needs a `Results` to add the notification token.
Also, the idea is against our ownership rules of listeners.
```java
query.addListener(new Listener(...));
```

So when should the anonymous listener be GCed? When the query gets GCed?

It really adds some unnecessary things to our API especially `query.findAll().asObservable()` will just do the job.

**Comment by beeender** [comment]
@cmelchior shall we close this since `RealmQuery` doesn't present the results set and it shouldn't be used in this way.
